src/command_modules/azure-cli-eventgrid/azure/cli/command_modules/eventgrid/_params.py

```diff
@@ -47,21 +48,27 @@ def load_arguments(self, _):
     with self.argument_context('eventgrid topic') as c:
         c.argument('topic_name', arg_type=name_type, help='Name of the topic', id_part='name', completer=get_resource_name_completion_list('Microsoft.EventGrid/topics'))
 
+    with self.argument_context('eventgrid topic key list') as c:
+        c.argument('topic_name', arg_type=name_type, help='Name of the topic', id_part=None, completer=get_resource_name_completion_list('Microsoft.EventGrid/topics'))
+
     with self.argument_context('eventgrid event-subscription') as c:
-        c.argument('topic_name', help='Name of the Event Grid topic', options_list=['--topic-name'], completer=get_resource_name_completion_list('Microsoft.EventGrid/topics'))
+        c.argument('topic_name', deprecate_info=c.deprecate(expiration='2.1.0', hide=True), help='Name of Event Grid topic', options_list=['--topic-name'], completer=get_resource_name_completion_list('Microsoft.EventGrid/topics'))
```

Comment:
@kalyanaj I believe the reason we decided to split the argument in the beginning is because specifying resource ids is not consistent with most commands in the CLI. 
@tjprescott to comment.

Reply:
Thanks @williexu. To provide additional background/context here, we have a few different categories of resource types (eventgrid topic, Azure subscription, resource group, other Azure resources such as storage account, eventgrid domain ) to which you can create an event subscription, and per discussion with @tjprescott a few months back we landed on having a single parameter --source-resource-id that can be used to specify the resource ID to which the event subscription is being created. The alternative way of having different set of parameters was getting unwieldy/hard to understand which is why this new parameter has been introduced.

Reply:
I'm thinking you should still redirect them here to the new parameter. My only comment on the new parameter is to drop the `-id` part and just call it `source-resource`.